### PR TITLE
Fix up some of the API conventions docs

### PR DESCRIPTION
**Pull request by thockin**
Make it clearer wrt annotations and prefixes.

### CODE REVIEW

File: contributors/devel/api-conventions.md [hunk 1]
```diff
@@ -1251,29 +1251,38 @@ to, but if unspecified, it could be set by default, such as to the resource
...
+primarily generated and consumed by tooling and system extensions, or are used
+by end-users to engage non-standard behavior of components.  For example, an
+annotation might be used to indicate that an instance of a resource expects
+additional handling by non-kubernetes controllers. Annotations may carry
+arbitrary payloads, including JSON documents.  Like labels, annotation keys can
+be prefixed with a governing domain (e.g. `example.com/key-name`).  Unprefixed
+keys (e.g. `key-name`) are reserved for end-users.  Third-party components must
+use prefixed keys.  Key prefixes under the "kubernetes.io" and "k8s.io" domains
+are reserved for use by th ekubernetes project and must not be used by
+thrid-parties.
```
Comment (cblecker) [inline contributors/devel/api-conventions.md]:
```
contributors/devel/api-conventions.md:1264:0: "thrid" is a misspelling of "third"
```

File: contributors/devel/api-conventions.md [hunk 2]
```diff
@@ -1251,29 +1251,38 @@ to, but if unspecified, it could be set by default, such as to the resource
...
+Annotations have very different intended usage from labels. They are
+primarily generated and consumed by tooling and system extensions, or are used
+by end-users to engage non-standard behavior of components.  For example, an
+annotation might be used to indicate that an instance of a resource expects
+additional handling by non-kubernetes controllers. Annotations may carry
+arbitrary payloads, including JSON documents.  Like labels, annotation keys can
+be prefixed with a governing domain (e.g. `example.com/key-name`).  Unprefixed
+keys (e.g. `key-name`) are reserved for end-users.  Third-party components must
+use prefixed keys.  Key prefixes under the "kubernetes.io" and "k8s.io" domains
+are reserved for use by th ekubernetes project and must not be used by
```
Comment (krmayankk) [inline contributors/devel/api-conventions.md]: nit: `by the kubernetes`

File: contributors/devel/api-conventions.md [hunk 3]
```diff
@@ -1251,29 +1251,38 @@ to, but if unspecified, it could be set by default, such as to the resource
...
+thrid-parties.
+
+In early versions of Kubernets, some in-development features represented new
+API fields as annotations, generally with the form `something.alpha.kubernetes.io/name` or
+`something.beta.kubernetes.io/name` (depending on our confidence in it). This
+pattern is deprecated.  Some such annotations may still exist, but no new
+annotations may be defined.
```
Comment (krmayankk) [inline contributors/devel/api-conventions.md]: Would it be helpful to also mention where these in-development features API fields are moving ?

File: contributors/devel/api-conventions.md (hunk 2 above)
Comment (thockin) [reply contributors/devel/api-conventions.md]: ACK

File: contributors/devel/api-conventions.md (hunk 1 above)
Comment (thockin) [reply contributors/devel/api-conventions.md]: ACK

File: contributors/devel/api-conventions.md (hunk 3 above)
Comment (thockin) [reply contributors/devel/api-conventions.md]: ACK

**Comment by thockin** [comment]
re-pushed

File: contributors/devel/api-conventions.md [hunk 4]
```diff
@@ -1251,29 +1251,38 @@ to, but if unspecified, it could be set by default, such as to the resource
...
-newer alpha/beta API versions in the older stable storage version, may be
-represented as annotations with the form `something.alpha.kubernetes.io/name` or
-`something.beta.kubernetes.io/name` (depending on our confidence in it). For
-example `net.alpha.kubernetes.io/policy` might represent an experimental network
-policy field. The "name" portion of the annotation should follow the below
-conventions for annotations. When an annotation gets promoted to a field, the
-name transformation should then be mechanical: `foo-bar` becomes `fooBar`.
+Annotations have very different intended usage from labels. They are
+primarily generated and consumed by tooling and system extensions, or are used
+by end-users to engage non-standard behavior of components.  For example, an
```
Comment (liggitt) [inline contributors/devel/api-conventions.md]: "engage non-standard behavior of components" reads strangely, consider rephrasing

**Comment by liggitt** [comment]
/lgtm
/hold

one nit, up to you whether to rephrase or not. unhold at will

**Comment by bgrant0607** [comment]
/hold cancel
Wording can be improved in a followup

**Pull request merged**